src/Route.php

```diff
@@ -17,25 +17,60 @@ class Route
     /** @var mixed */
     public $handler;
 
+    public bool $isStatic = false;
+
     /**
      * @param mixed   $handler
      * @param mixed[] $variables
      */
-    public function __construct(string $httpMethod, $handler, string $regex, array $variables)
+    public function __construct(string $httpMethod, $handler, string $regex, array $variables, bool $isStatic = false)
     {
         $this->httpMethod = $httpMethod;
         $this->handler = $handler;
         $this->regex = $regex;
         $this->variables = $variables;
+        $this->isStatic = $isStatic;
     }
 
     /**
      * Tests whether this route matches the given string.
+     *
+     * @param string $string URI string to match
      */
-    public function matches(string $str): bool
+    public function matches(string $string): bool
     {
+        if ($this->isStatic) {
+            return $string === $this->regex;
+        }
+
         $regex = '~^' . $this->regex . '$~';
 
-        return (bool) preg_match($regex, $str);
+        return (bool) preg_match($regex, $string);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function handler()
+    {
+        return $this->handler;
+    }
+
+    public function regex(): string
+    {
+        return $this->regex;
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function variables(): array
+    {
+        return $this->variables;
+    }
+
+    public function isStatic(): bool
+    {
+        return $this->isStatic;
```

Comment:
We should not add this IMHO as this is a DTO IMHO and the performance trade-offs are relevant for this library.

```suggestion
```

Reply:
The route object should actually describe the route in my opinion.